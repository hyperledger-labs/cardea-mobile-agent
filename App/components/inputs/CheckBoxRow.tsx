import React from 'react'
import { View, StyleSheet, TouchableOpacity, Text } from 'react-native'
import Icon from 'react-native-vector-icons/MaterialIcons'

import { Colors, TextTheme } from '../../theme'

interface Props {
  title: string
  accessibilityLabel?: string
  checked: boolean
  onPress: () => void
}

const style = StyleSheet.create({
  container: {
    flex: 1,
    flexDirection: 'row',
    alignItems: 'center',
    margin: 10,
  },
  text: {
    flexShrink: 1,
    ...TextTheme.normal,
    marginLeft: 10,
  },
})

const CheckBoxRow: React.FC<Props> = ({ title, accessibilityLabel, checked, onPress }) => {
  const accessible = accessibilityLabel && accessibilityLabel !== '' ? true : false

  return (
    <View style={style.container}>
      <TouchableOpacity accessible={accessible} accessibilityLabel={accessibilityLabel} onPress={onPress}>
        {checked ? (
          <Icon name={'check-box'} size={36} color={Colors.primary} />
        ) : (
          <Icon name={'check-box-outline-blank'} size={36} color={Colors.primary} />
        )}
      </TouchableOpacity>
      <Text style={[style.text]}>{title}</Text>
    </View>
  )
}

export default CheckBoxRow
