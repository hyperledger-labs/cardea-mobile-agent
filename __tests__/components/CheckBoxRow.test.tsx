import React from 'react'
import { create } from 'react-test-renderer'

import CheckBoxRow from '../../App/components/inputs/CheckBoxRow'

jest.mock('react-native-vector-icons/MaterialIcons', () => 'Icon')

describe('CheckBoxRow Component', () => {
  it('Renders correctly', () => {
    const tree = create(
      // eslint-disable-next-line
      <CheckBoxRow title={'Hello Fried'} accessibilityLabel={'Hey'} checked={true} onPress={() => {}} />
    ).toJSON()

    expect(tree).toMatchSnapshot()
  })
})
