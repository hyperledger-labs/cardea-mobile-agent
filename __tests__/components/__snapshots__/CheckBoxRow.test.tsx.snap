// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CheckBoxRow Component Renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "flex": 1,
      "flexDirection": "row",
      "margin": 10,
    }
  }
>
  <View
    accessibilityLabel="Hey"
    accessible={true}
    collapsable={false}
    focusable={true}
    nativeID="animatedComponent"
    onClick={[Function]}
    onResponderGrant={[Function]}
    onResponderMove={[Function]}
    onResponderRelease={[Function]}
    onResponderTerminate={[Function]}
    onResponderTerminationRequest={[Function]}
    onStartShouldSetResponder={[Function]}
    style={
      Object {
        "opacity": 1,
      }
    }
  >
    <Icon
      color="#35823F"
      name="check-box"
      size={36}
    />
  </View>
  <Text
    style={
      Array [
        Object {
          "color": "#FFFFFF",
          "flexShrink": 1,
          "fontSize": 18,
          "fontWeight": "normal",
          "marginLeft": 10,
        },
      ]
    }
  >
    Hello Fried
  </Text>
</View>
`;
